arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1113,6 +1113,64 @@ std::shared_ptr<LogicalView> ClusterInfo::getView(
   return nullptr;
 }
 
+std::shared_ptr<LogicalView> ClusterInfo::getViewCurrent(
+    DatabaseID const& databaseID,
+    ViewID const& viewID
+) {
+  static const auto lookupView = [](
+      AllViews const& dbs,
+      DatabaseID const& databaseID,
+      ViewID const& viewID
+  ) noexcept -> std::shared_ptr<LogicalView> {
+    auto const db = dbs.find(databaseID); // look up database by id
+
+    if (db != dbs.end()) {
+      auto& views = db->second;
+      auto const itr = views.find(viewID); // look up view by id (or by name)
+
+      if (itr != views.end()) {
+        return itr->second;
+      }
+    }
+
+    return nullptr;
+  };
+
+  if (std::this_thread::get_id() == _planLoader) {
```

Comment:
is this really a good idea ? Does the validity of this ID depend on the way the thread was created  (i.e. are we using `std::thread` everywhere now )?

Reply:
AFAIK std::this_thread::get_id() doesn't depend on the usage of std::thread

Reply:
There is no guarantee that it must work with non std::threads. But as the std::thread stuff is just a thin wrapper around the native stuff it is very likely to work.

Reply:
http://eel.is/c++draft/thread.threads#thread.thread.id-1